libs/common/src/test-helpers/mock-data/utilisation-report.entity.mock-builder.ts

```diff
@@ -71,6 +71,11 @@ export class UtilisationReportEntityMockBuilder {
     return this;
   }
 
+  public withoutAzureFileInfo(): UtilisationReportEntityMockBuilder {
+    this.report.azureFileInfo = undefined;
+    return this;
+  }
```

Comment:
My main problem with `.withAzureFileInfo(azureFileInfo: AzureFileInfoEntity | undefined)` is that calling `.withAzureFileInfo()` will remove the azure file info entity currently attached to the object, but we have done that by calling a method which says "with". It would be like asking for a burger with ketchup, being served a burger without ketchup and being told "you didn't specify how much ketchup, so I assumed you meant 0" 😅 

@ChristianMcc another one where it may just be better to get your opinion. The two options we have come up with are (inside `UtilisationReportEntityMockBuilder`)
```typescript
withAzureFileInfo(azureFileInfo: AzureFileInfoEntity | undefined) {
  this.report.azureFileInfo = azureFileInfo;
}
```
or
```typescript
withAzureFileInfo(azureFileInfo: AzureFileInfoEntity) {
  this.report.azureFileInfo = azureFileInfo;
}

withoutAzureFileInfo() {
  this.report.azureFileInfo = undefined;
}
```

Reply:
Just to be clear:
<img width="1124" alt="Screenshot 2024-03-14 at 15 47 49" src="https://github.com/UK-Export-Finance/dtfs2/assets/67907735/148cc67d-9751-4cfd-aa97-1c0e0ef539ce">

Reply:
Ah nice - I didn't realise typescript would do that. In that case, let's go with `withAzureFileInfo` and I'll update that in this PR